### PR TITLE
Font styling, child windows and images.

### PR DESCRIPTION
**Pull request by dlannan**
I have added font styling (within a window) using the font_push and font_pop methods.
Fonts can be loaded via file or data (using resource load).
Child frames can be given window flags so that parts of a window can change behavior.
Images can be loaded using files or data (using resource load).
image

### CODE REVIEW

**Comment by britzl** [comment]
Ha! I was adding image loading via stb_image.h the other day, but didn't have a chance to wrap it up and commit! You beat me to it! :-)

File: imgui/src/imgui/base64.c [hunk 1]
```diff
@@ -0,0 +1,119 @@
+/*
```
Comment (britzl) [inline imgui/src/imgui/base64.c:1]: We have functions for b64 encode/decode in the Defold SDK: 

https://defold.com/ref/stable/dmCrypt/#dmCrypt::Base64Encode:src-src_len-dst-dst_len[in,out]

We might as well use those?

File: imgui/src/imgui/imgui_impl_metal.h [hunk 2]
```diff
@@ -0,0 +1,28 @@
+// dear imgui: Renderer Backend for Metal
```
Comment (britzl) [inline imgui/src/imgui/imgui_impl_metal.h:1]: Should this be included? It is not uses right?

**Review by britzl — CHANGES_REQUESTED** [review]

File: imgui/src/imgui/base64.c (hunk 1 above)
Comment (dlannan-fmad) [reply imgui/src/imgui/base64.c:1]: Ahh, soz. Yes we should. Will make a change and test.

File: imgui/src/imgui/imgui_impl_metal.h (hunk 2 above)
Comment (dlannan-fmad) [reply imgui/src/imgui/imgui_impl_metal.h:1]: Will check. This may have come in from a previous version I was playing with.

**Comment by dlannan-fmad** [comment]
> Ha! I was adding image loading via stb_image.h the other day, but didn't have a chance to wrap it up and commit! You beat me to it! :-)

Soz :) .. really wanted some images in the web rendering :)

File: imgui/src/imgui/base64.c (hunk 1 above)
Comment (dlannan) [reply imgui/src/imgui/base64.c]: Have added dmCrypt Base64Decode

File: imgui/src/imgui/imgui_impl_metal.h (hunk 2 above)
Comment (dlannan) [reply imgui/src/imgui/imgui_impl_metal.h]: Removed files.

File: imgui/src/extension_imgui.cpp [hunk 3]
```diff
@@ -15,20 +15,207 @@
...
+
+    ImgObject     iobj;
+    unsigned char *strdata = (unsigned char *)luaL_checkstring(L, 2);
+    int lendata = luaL_checkinteger(L, 3);
+    iobj.data = stbi_load_from_memory( strdata, lendata, &iobj.w, &iobj.h, NULL, STBI_rgb_alpha);
+    //printf("Loaded Image: %s %d %d \n", filename, iobj.w, iobj.h);
+    
+    if(iobj.data == nullptr)
+    {
+        printf("Error loading image: %s\n", filename);
```
Comment (britzl) [inline imgui/src/extension_imgui.cpp:122]: Let's use dmLogError() instead!

File: imgui/src/extension_imgui.cpp [hunk 4]
```diff
@@ -15,20 +15,207 @@
...
+            lua_pushinteger(L, i);
+            return 1;
+        }
+    }
+
+    ImgObject     iobj;
+    iobj.data = stbi_load(filename, &iobj.w, &iobj.h, NULL, STBI_rgb_alpha);
+    if(iobj.data == nullptr)
+    {
+        printf("Error loading image: %s\n", filename);
```
Comment (britzl) [inline imgui/src/extension_imgui.cpp:160]: dmLogError()

**Comment by britzl** [comment]
Added a comment about printf vs dmLogError.

Please also add the new functions to the api definition (for autocomplete etc):

https://github.com/britzl/extension-imgui/blob/master/imgui/api/imgui.script_api

(I think I've forgotten to add some functions there as well, I'll check this!)

**Comment by dlannan-fmad** [comment]
No probs. Yes, I havent updated the api list :) .. will do so.

File: imgui/src/extension_imgui.cpp (hunk 3 above)
Comment (dlannan) [reply imgui/src/extension_imgui.cpp]: Done.

File: imgui/src/extension_imgui.cpp (hunk 4 above)
Comment (dlannan) [reply imgui/src/extension_imgui.cpp]: Done.

**Comment by dlannan** [comment]
Updating api. Will make sure everything is in there.

**Comment by dlannan** [comment]
Ok. API docs should be all up to date now. Let me know if I missed anything.

**Review by britzl — APPROVED** [review]

**Comment by britzl** [comment]
Thanks!

**Pull request merged**